zasilkovna.php

```diff
@@ -1260,43 +1260,57 @@ public function getTailBlockHtml(
     private function addPacketaToolbarButtons(\VirtueMartModelZasilkovna\Order\Order $order)
     {
         $toolbar = JToolbar::getInstance('toolbar');
+        $isJoomla4 = version_compare(JVERSION, '4.0', 'ge');
+        if ($isJoomla4) {
+            if ($order->hasPacketId()) {
+                $button = new LinkButton(
+                    'packetaPrintLabel',
+                    JText::_('PLG_VMSHIPMENT_PACKETERY_PRINT_LABEL'),
+                    [
+                        'icon' => 'icon-printer',
+                        'class' => 'btn btn-small',
+                        'url' => '#',
+                    ]
+                );
 
-        if ($order->hasPacketId()) {
-            $button = new LinkButton(
-                'packetaPrintLabel',
-                JText::_('PLG_VMSHIPMENT_PACKETERY_PRINT_LABEL'),
-                [
-                    'icon' => 'icon-printer',
-                    'class' => 'btn btn-small',
-                    'url' => '#',
-                ]
-            );
-
-        } else {
-            $button = new LinkButton(
-                'packetaSubmitPacket',
-                JText::_('PLG_VMSHIPMENT_PACKETERY_SUBMIT_TO_PACKETA'),
-                [
-                    'icon' => 'icon-envelope',
-                    'class' => 'btn btn-small',
-                    'url' => 'index.php?option=com_virtuemart&view=zasilkovna&task=submitPacket&virtuemart_order_id=' . $order->getVirtuemartOrderId(),
-                ]
-            );
-        }
+                $button->setParent($toolbar);
+            } else {
+                $submitUrl = 'index.php?option=com_virtuemart&view=zasilkovna&task=submitPacket&virtuemart_order_id=' . $order->getVirtuemartOrderId();
+                $button = new LinkButton(
+                    'packetaSubmitPacket',
+                    JText::_('PLG_VMSHIPMENT_PACKETERY_SUBMIT_TO_PACKETA'),
+                    [
+                        'icon' => 'icon-envelope',
+                        'class' => 'btn btn-small',
+                        'url' => $submitUrl,
+                    ]
+                );
 
-        $button->setParent($toolbar);
+                $button->setParent($toolbar);
```

Comment:
Tento řádek je v obou větvích, stačí jednou a zkrátí se to.

Reply:
Jeden radek s kodem odstranen a kod vysunut.